Ask me when incremental (#558)
dataform-co/dataform: content/docs/how-to-guides/incremental-datasets.md

SELECT timestamp, action
FROM weblogs.user_actions

${ when(incremental(), `WHERE timestamp > (SELECT MAX(timestamp) FROM ${self()})` }
```

First the script sets the type of the dataset to `incremental`.

> It then specifies a `WHERE` clause using a `when(incremental(), ...)` in-line:

Reviewer: this is odd english; "in-line" is not a noun, there can't be "a" "in-line". 

```
It then specifies a `WHERE` clause using the `when()` and `incremental()` functions:
```

Since incremental datasets are usually timestamp based, it's a best practice to set up dataset partioning on the timestamp field to speed up downstream queries.

For more information, check out the [BigQuery](warehouses/bigquery) and [Redshift](warehouses/redshift) guides.

## Conditional code if incremental

> For functionality other than a `WHERE` clause, `when()` combined with `incremental()` can be used. For example if there was a separate field `weblogs.user_alternative_actions` containing an alternative set of user events, and it was desirable for the script to use this only when incremental, then a neat way to manage this would be to use:

Reviewer: ```suggestion
For functionality other than a `WHERE` clause, `when()` combined with `incremental()` can be used. For example, assuming there is a field `weblogs.user_alternative_actions` containing an alternative set of user events, and it is desirable for the script to use this only when running incrementally, you could write:
```

```js
config { type: "incremental" }

SELECT timestamp, action
FROM ${ when(incremental(), weblogs.user_alternative_actions, weblogs.user_actions) }

> incremental_where {

Reviewer: IMO we should get rid of all mention of `incremental_where` from the docs.

Author: Sounds good to me
